### ISSUE TITLE
Less builder hangs when compiling empty files

### ISSUE THREAD

**Issue by Gagaro** [plone/plone.app.theming#72]
If I try to compile an empty less file (or a file which will render to an empty css file such as one with only comments), it's stuck at `Working....`.

**Comment by MrTango** [comment]
LESS Builder does not work at all, except for the barceloneta less file /less/barceloneta.plone.less.
If i add a new theme and try to compile a new less file, it says working for ever but nothing happen ;).
I can see an 404 error in the inspector, when trying to get the ++theme++myba/@@plone.resourceeditor.filemanager-actions view.

Also it is not clear where the default value "placeholder" for the compiled file name in the case of barceloneta-compiled.css comes from. Where can i define this default?

**Comment by ebrehault** [comment]
@MrTango it does work with any theme, you just need to declare the resources locations in your manifest.cfg.
See https://community.plone.org/t/easy-plone-5-theme-customizing/944/10?u=ebrehault

**Comment by vangheem** [comment]
Less building should work on any file also. The UI could use a bit of work maybe but we've been using it for a while without any issues.

Any there errors in the log?

\cc @obct537

**Comment by MrTango** [comment]
@ebrehault thx for advice, i'll try that. But how can i set this TTW? I just created a new theme TTW ;).
@vangheem no the less building from the theme editor does not work with any less files. Do you mean building from the resource registry?

**Comment by vangheem** [comment]
heh, well, like I said, it's been working for us: https://www.dropbox.com/s/cdq84nwbqsvwero/less-compiling.mov?dl=0

**Comment by vangheem** [comment]
Look to see if there are errors in the log. I imagine there needs to be better error reporting.

**Comment by obct537** [comment]
@vangheem @MrTango I've been looking at this, and LESS building does definitely seem broken in the most recent build.

**Comment by ebrehault** [comment]
@MrTango sure, you create your theme TTW, and you just edit its manifest.cfg using the theming editor

**Comment by MrTango** [comment]
@ebrehault ah yes, your are right ;)

**Comment by obct537** [comment]
@vangheem @MrTango @ebrehault This PR should fix this issue: https://github.com/plone/plone.resourceeditor/pull/12

**Comment by obct537** [comment]
I stand corrected. It fixes the probably causing the LESS builder to fail all together, but the original issue with empty files still seems to be there

**Comment by vangheem** [comment]
I finally see what the problem was. If the file is missing that you're trying to save to, it'll throw a NotFound error from the previous code. Building should work if the file exists though.

**Comment by obct537** [comment]
Yeah, that's essentially what the PR I made addresses. What I don't fully understand is why this just now started happening?
